http/src/main/java/io/micronaut/http/HttpHeaders.java

```diff
@@ -722,4 +767,15 @@ default Optional<String> getAuthorization() {
     default Optional<String> getContentType() {
         return findFirst(CONTENT_TYPE);
     }
+
+    /**
+     * Equivalent to {@code contains(SERVER)}, optimized for internal use only.
+     *
+     * @return {@code contains(SERVER)}
+     * @since 4.0.0
+     */
+    @Internal
+    default boolean containsServer() {
```

Comment:
rather than include an internal method into the public API which is not ideal just cast down to NettyHttpRequest and invoke the logic directly

Reply:
it's called from RouteExecutor which is in http-server so it can't access netty

Reply:
code should be refactored then to allow subclass RouteExecutor for the Netty implementation. There shouldn't be internal methods like this in public interfaces. If we absolutely must keep it, then it should be renamed `containsServerHeader`

Reply:
i think i'll just revert it then, it's a really tiny perf difference